metapackages/auto-instrumentations-node/src/utils.ts

```diff
@@ -61,91 +62,87 @@ import {
+    config: { enabled: true } as MongoDBInstrumentationConfig,
+  },
+  // '@opentelemetry/instrumentation-mysql': {
+  //   instrumentation: MySQLInstrumentation,
+  //   config: { enabled: true } as MySQLInstrumentationConfig,
+  // },
+  '@opentelemetry/instrumentation-pg': {
+    instrumentation: PgInstrumentation,
+    config: { enabled: true } as PgInstrumentationConfig,
+  },
+  // '@opentelemetry/instrumentation-redis': {
+  //   instrumentation: RedisInstrumentation,
+  //   config: { enabled: true } as RedisInstrumentationConfig,
+  // },
 };
 
-const defaultConfigs: InstrumentationConfigMap = {
-  '@opentelemetry/instrumentation-dns': { enabled: true },
-  '@opentelemetry/instrumentation-express': { enabled: true },
-  '@opentelemetry/instrumentation-http': { enabled: true },
-  '@opentelemetry/instrumentation-graphql': { enabled: true },
-  '@opentelemetry/instrumentation-grpc': { enabled: true },
-  '@opentelemetry/instrumentation-koa': { enabled: true },
-  '@opentelemetry/instrumentation-ioredis': { enabled: true },
-  '@opentelemetry/instrumentation-mongodb': { enabled: true },
-  // '@opentelemetry/instrumentation-mysql': { enabled: true },
-  '@opentelemetry/instrumentation-pg': { enabled: true },
-  // '@opentelemetry/instrumentation-redis': { enabled: true }
+type InstrumentationConfigMap = {
+  [Name in keyof typeof InstrumentationMap]?: typeof InstrumentationMap[Name]['config'];
 };
 
 export function getNodeAutoInstrumentations(
   inputConfigs: InstrumentationConfigMap = {}
 ): Instrumentation[] {
-  const configs: InstrumentationConfigMap = Object.assign(
-    {},
-    defaultConfigs,
-    inputConfigs
-  );
-  const keys = Object.keys(configs);
+  for (const name of Object.keys(inputConfigs)) {
+    if (!Object.prototype.hasOwnProperty.call(InstrumentationMap, name)) {
+      diag.error(`Provided instrumentation name "${name}" not found`);
+      continue;
+    }
+  }
+
   const instrumentations: Instrumentation[] = [];
-  keys.forEach(key => {
-    const Instance = InstrumentationMap[key as keyof InstrumentationConfigMap];
-    const config = configs[key as keyof InstrumentationConfigMap];
-    if (Instance) {
-      try {
-        if (config?.enabled === false) {
-          diag.debug(`Disabling instrumentation for ${key}`);
-        } else {
-          diag.debug(`Loading instrumentation for ${key}`);
-          instrumentations.push(new Instance(config));
-        }
-      } catch (e) {
-        diag.error(e);
-      }
-    } else {
-      diag.error(`Provided instrumentation name "${key}" not found`);
+
+  for (const name of Object.keys(InstrumentationMap) as Array<
+    keyof typeof InstrumentationMap
+  >) {
+    const Instance = InstrumentationMap[name].instrumentation;
+
+    const userConfig = inputConfigs[name] ?? { enabled: true };
+
+    if (userConfig.enabled === false) {
+      diag.debug(`Disabling instrumentation for ${name}`);
+      continue;
     }
-  });
+
+    try {
```

Comment:
@dyladan you have missed quite important thing, the user config should be merged with default instrumentation config

Reply:
Fixed

Reply:
Originally I did not merge them because all the defaults were just the same so it didn't seem like it mattered. I fixed it in case we want to provide defaults in the future but I really think default should be defined in the plugin not in the metapackage